scripts/train_lm.py

```diff
@@ -146,12 +146,39 @@ def WriteMetaparameters(metaparameters, ngram_order, num_train_sets, out_file):
 
 def FormatMetaparameters(metaparameters):
     out = []
+    # get the maximum length of decimals of metaparameters
+    decimal_length = 0
     for param in metaparameters:
-        x = '{:.3f}'.format(param)
-        if x == '0.00':
-            x = '{:.3g}'.format(param)
-        out.append(x)
-
+        temp_length = len(str(param))
+        if decimal_length < temp_length:
+            decimal_length = temp_length
+    decimal_length -= 2
+    
+    temp_original = metaparameters[0]
+    temp = round(temp_original, 3)
+    for param in metaparameters[1:]:
+        round_index = 3
+        x_original = param
+        x = round(x_original, round_index)
+        while(temp == x and round_index <= decimal_length):
```

Comment:
I think there are two possible issues
First,
input = [0.9999999,0.99999999999,0.999999999999999,0.9999999999999999999]
output: train_lm.py: there is a 1.0. metaparameters should be in range (0, 1).
(where there should be no 1.0)

Second,
input = [0.0000012344,0.00000123456,0.000001234567]
output = "0.000001234,0.000001235,0.000001"
the last thing 0.000001 only has 1 significant digit, which may not be what we desire (we want at least three significant digit)